source/vir/src/sst_to_air.rs

```diff
@@ -1471,21 +1471,23 @@ fn stm_to_stmts(ctx: &Ctx, state: &mut State, stm: &Stm) -> Result<Vec<Stmt>, Vi
             }
             let mut ens_args: Vec<_> =
                 typ_args.into_iter().chain(ens_args_wo_typ.into_iter()).collect();
-            if let Some(Dest { dest, is_init }) = dest {
```

Comment:
Actually, I do wanna double-check I understand this right:

It looks like in this situation, the SST/AIR creates a 'dest' variable, but if there's no return-value, then we skip this whole chunk of code, which means we end up not assigning to the variable in AIR.

But that's fine, because it's a unit variable, and the type invariant exists, so it's only constrained to be that one value anyway.

Is that right?

Reply:
That's my understanding too, yes.
However, now that you mention it, skipping havoc-ing here is only sound if this case can only happen if the destination variable is unit (rather, `tuple%0` here). We do rely on the type checker for this. We could conceivably always havoc in this case, as a defensive strategy.